### ISSUE TITLE
Removed Mesos support?

### ISSUE THREAD

**Issue by Spritekin** [docker/classicswarm#2859]
Gentlemen,

I write this message to disagree about the removal of the Mesos integration.

Right now, the company I work for has Swarm running on top of a Mesos cluster for over a year. In that time Swarm has executed an average of 50k jobs per day and millions of jobs per month. It runs steadilly and reliably.

Granted, I have done many adjustments to the Swarm code that I have posted here multiple times only to fail on the builds with no apparent reason or rejected because they were misunderstood.

Still in my posted issues I write about how to fix many of the official Swarm-Mesos shortcomings like:
- job resilience to swarm crashes
- improved swarm health checks
- fractional CPU allocation, using shares instead of full CPUs.
- mounting folders from the cluster in a container
- running a container in privileged mode

So in other words, I would have shared all this changes if the build pipeline was more robust, but to be honest, every time I posted a solution I had to resubmit like 10 times just to fail again and again or just get rejected.

I imagine there might be a business reason to drop Mesos and I can't blame you for that. Still its regretful so I will probably just post my branch in some repository so this feature is maintained.

Regards,
Guillermo Rodriguez

**Comment by nishanttotla** [comment]
@Spritekin thank you for sharing your feedback, we appreciate it.

The key reasons for removing the Mesos integration were the following:
1. Lack of maintainership: while we would have liked to keep the integration, it has not kept up with the latest offerings of Mesos, and the rate at which we have been able to fix issues has been too low to reliably recommend the integration to new users.
2. Lack of usage: given that the code base was not maintained and kept up to date, we tried to understand the usage and requirements, and did not get a strong signal that finding ways to build maintainership would make sense for the project at this point.

There are no business reasons in particular, because we were happy to add the Mesos integration and it did draw usage when it was added.

cc @vieux @allencloud @chungers
